homeassistant/components/recorder/core.py

```diff
@@ -553,10 +553,10 @@ def _adjust_lru_size(self) -> None:
         If the number of entities has increased, increase the size of the LRU
         cache to avoid thrashing.
         """
-        new_size = self.hass.states.async_entity_ids_count() * 2
-        self.state_attributes_manager.adjust_lru_size(new_size)
-        self.states_meta_manager.adjust_lru_size(new_size)
-        self.statistics_meta_manager.adjust_lru_size(new_size)
+        if new_size := self.hass.states.async_entity_ids_count() * 2:
```

Comment:
This could actually set the size to 0 if there was nothing in the state machine. This probably would never happen